sycl/test/fpga_tests/fpga_pipes.cpp

```diff
@@ -0,0 +1,205 @@
+// RUN: %clang -std=c++11 -fsycl %s -o %t.out -lstdc++ -lOpenCL -lsycl
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+//==------------- fpga_pipes.cpp - SYCL FPGA pipes test --------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+#include <iostream>
+
+// For pipes created with namespaces set
+namespace some {
+  class pipe;
+}
+
+using namespace cl::sycl;
+
+const unsigned dataSize = 1024;
+
+int main() {
+  int data[dataSize];
+  for (size_t i = 0; i != dataSize; ++i)
+    data[i] = 0;
+
+  {
+    // Test for non-blocking pipes
+    queue Queue;
+    using Pipe = pipe<class some_pipe, int, 1>;
+
+    Queue.submit([&](handler& cgh) {
+      cgh.parallel_for<class foo_nb>(range<1> { dataSize }, [=](id<1> idx) {
+        bool SuccessCode = false;
+        while (!SuccessCode)
+          Pipe::write(42, SuccessCode);
```

Comment:
The current API is overly verbose. This should be done in 1 line:
```
while (!Pipe::write(42)) ;
```

Reply:
One liners are expected for blocking pipe. The code example above is about non-blocking pipes.

Reply:
See line 180:

Reply:
Sure, that is my point. You have to find with @mkinsner how the API can lead to a 1-liner here too. 
The idea behind SYCL is to be simpler to use than OpenCL... :-)

Reply:
Ah, I see your point. Just missed while in the beggining.